ibc-core/ics02-client/context/src/context.rs

```diff
@@ -4,17 +4,34 @@ use ibc_core_host_types::identifiers::ClientId;
 use ibc_core_host_types::path::{ClientConsensusStatePath, ClientStatePath};
 use ibc_primitives::Timestamp;
 
-use super::client_state::ClientState;
-use super::consensus_state::ConsensusState;
+use crate::client_state::{ClientStateExecution, ClientStateValidation};
+use crate::consensus_state::ConsensusState;
 
 /// Defines the methods available to clients for validating client state
 /// transitions. The generic `V` parameter in
 /// [crate::client_state::ClientStateValidation] must
 /// inherit from this trait.
-pub trait ClientValidationContext {
-    /// Returns the time and height when the client state for the given
-    /// [`ClientId`] was updated with a header for the given [`Height`]
-    fn update_meta(
+pub trait ClientValidationContext: Sized {
+    type ClientStateRef: ClientStateValidation<Self>;
+    type ConsensusStateRef: ConsensusState;
+
+    /// Returns the ClientState for the given identifier `client_id`.
+    ///
+    /// Note: Clients have the responsibility to store client states on client creation and update.
+    fn client_state(&self, client_id: &ClientId) -> Result<Self::ClientStateRef, ContextError>;
+
+    /// Retrieve the consensus state for the given client ID at the specified
+    /// height.
+    ///
+    /// Returns an error if no such state exists.
+    ///
+    /// Note: Clients have the responsibility to store consensus states on client creation and update.
+    fn consensus_state(
+        &self,
+        client_cons_state_path: &ClientConsensusStatePath,
+    ) -> Result<Self::ConsensusStateRef, ContextError>;
+
+    fn client_update_meta(
```

Comment:
I just renamed this particular method and added a `client` prefix since I found the alone `update_meta` a bit confusing. As this method is newly introduced (post v0.50.0) and has not been included in any release yet, I thought it would be ok to touch it here. But to move forward with your suggestion and consistently updating namings we should touch on other relevant methods and types too. If you agree (since it will impose another group of breaking changes) let’s address it in a later PR.

Reply:
Sounds ok with me :slightly_smiling_face: